apps/src/templates/sectionProgress/progressTables/ProgressTableContainer.jsx

```diff
@@ -46,7 +46,9 @@ class ProgressTableContainer extends React.Component {
     lessonOfInterest: PropTypes.number.isRequired,
     levelProgressByStudent: PropTypes.objectOf(
       PropTypes.objectOf(studentLevelProgressType)
-    ).isRequired
+    ).isRequired,
+    studentTimestamps: PropTypes.object.isRequired,
```

Comment:
Are these written to by other parts of Charlie's changes?  (It's ok if they are, just wanted to double-check.)

Reply:
These are props coming in through redux that were missing from propTypes, which I discovered while writing tests. By written to, did you mean passed into the component?

Reply:
@jamescodeorg this component subscribes to redux props needed by its children to avoid lots of duplicated work in `connect`. originally i was only validating prop types in the components that actually use the props, but we decided that it makes sense to validate pass-through props even though they're only needed in the children. does that make sense? and do you think this pattern (of validating pass-through props) makes sense?

Reply:
Oh, interesting.  I agree with this in principle, but in practice, does this mean that a component near the top of tree might end up validating a lot of properties?  Is there an accepted best practice around validating prop types in redux?  (To be honest, I still don't totally grok the principles behind redux.  Naively, it feels kind of like a giant global variable store with some data flow constraints and good debugging tools.)

Reply:
a quick search for best practice about this only produced this one not-very-definitive answer: https://stackoverflow.com/questions/49594216/should-i-use-propttypes-for-props-i-pass-through-a-react-component

tl;dr, one person says validate them, one person says don't, and one person offers another way to validate them by requiring `ChildComponent.propTypes`. since many of the pass-through props are used by multiple children, i don't think that last option makes sense here. so my vote is to leave this as-is.

@jamescodeorg your feeling about redux in our codebase is spot on. however, i believe that's just because we're not using it as intended. i met with Dave during my second week on the job to discuss this. unfortunately fixing this issue is going to require some major rearchitecting, but it's still my hope to start moving in that direction as part of our "north star" discussions during H1.

Reply:
Ok, thanks for doing the research.  I don't have a strong opinion (yet? :-) so whatever you think is best sounds good to me.